sentry-android-core/src/main/java/io/sentry/android/core/PerformanceAndroidEventProcessor.java

```diff
@@ -0,0 +1,46 @@
+package io.sentry.android.core;
+
+import io.sentry.EventProcessor;
+import io.sentry.protocol.MeasurementValue;
+import io.sentry.protocol.SentryTransaction;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/** Event Processor responsible for adding Android metrics to transactions */
+final class PerformanceAndroidEventProcessor implements EventProcessor {
+
+  private final boolean tracingEnabled;
+
+  private boolean sentStartMeasurement = false;
+
+  // transactions may be started in parallel, locking it so sentStartMeasurement
+  private final @NotNull Object measurementLock = new Object();
+
+  PerformanceAndroidEventProcessor(final @NotNull SentryAndroidOptions options) {
+    tracingEnabled = options.isTracingEnabled();
+  }
+
+  @Override
+  public @NotNull SentryTransaction process(
+      @NotNull SentryTransaction transaction, @Nullable Object hint) {
+    // the app start metric is sent only once when the 1st transaction happens
+    // after the app start is collected.
+    synchronized (measurementLock) {
+      if (!sentStartMeasurement && tracingEnabled) {
```

Comment:
yep, I forgot to commit that, I cant add directly to the transaction though, because when the span starts, I still don't know when the app start finishes.